ansible/roles/edu-sharing/tasks/custom_option.yml

```diff
@@ -0,0 +1,37 @@
+---
+
+# Implment custom_option task
+
+
+- name: Add custom Option 
+  block:
+   # first remove all custom options
+    - name: clear the custom options, before we add it.
+      xml:
+        path: '{{ edu_client_config_xml_path }}'
+        xpath: "/config/values/customOptions"
+        state: absent
+        pretty_print: yes
+      when: edu_custom_options | default([], true) | length>0
```

Comment:
if we just clear the existing values in case there is at least one value defined, then there is no possibility to remove all the custom-options. (for example: you may define customOptions and decide afterwards to not use it anymore -> then the config has to be cleared)

=> so, I would say, you should remove the condition here